src/include/com/amazonaws/kinesis/video/webrtcclient/Include.h

```diff
@@ -1273,6 +1292,13 @@ typedef struct {
     CHAR userName[MAX_ICE_CONFIG_USER_NAME_LEN + 1];                 //!< Username for the server
     CHAR password[MAX_ICE_CONFIG_CREDENTIAL_LEN + 1];                //!< Password for the server
 } IceConfigInfo, *PIceConfigInfo;
+
+typedef struct {
+    CHAR channelName[MAX_CHANNEL_NAME_LEN + 1]; //!< Signaling channel name, MUST SET
+    CHAR channelArn[MAX_ARN_LEN + 1];           //!< The arn of signaling channel, optional.
+    CHAR storageStreamArn[MAX_ARN_LEN + 1];     //!< The arn of kvs stream, optional if you already associate signaling channel with stream
+    BOOL storageStatus;                         //!< Indicate the association between channelArn and storageStreamArn. MUST SET
+} MediaStorageConfig, *PMediaStorageConfig;
```

Comment:
After reviewing the whole code, this struct seems unnecessary. The bool isn't needed, channel name and arn are already tracked, and stream arn could be added alone.
The SignalingSync shouldn't be modal.

Reply:
If we do not check the mode(media storage mode or p2p mode), we can remove it. However, I prefer adding some mechanism of checking the mode is correct as which customers use. In this way, we can prevent unwanted infinite loop inside the fsm unless you want to check streamArn in the joinStorageSession API.

Reply:
Updated this as the returned media storage configuration.